### ISSUE TITLE
Failed at the sqlite3@5.0.1 install script.

### ISSUE THREAD

**Issue by royeradames** [mapbox/node-sqlite3#1424]
This is the full error message I am getting when trying to install sqlite3.

$ npm i sqlite3

> sqlite3@5.0.1 install D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using request for node-pre-gyp https download
node-pre-gyp WARN Tried to download(403): https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v5.0.1/napi-v6-win32-x64.tar.gz
node-pre-gyp WARN Pre-built binaries not found for sqlite3@5.0.1 and node@14.15.1 (node-v83 ABI, unknown) (falling back to source compile with node-gyp)
gyp ERR! find VS
gyp ERR! find VS msvs_version not set from command line or npm config
gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
gyp ERR! find VS could not use PowerShell to find Visual Studio 2017 or newer
gyp ERR! find VS looking for Visual Studio 2015
gyp ERR! find VS - not found
gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
gyp ERR! find VS
gyp ERR! find VS **************************************************************
gyp ERR! find VS You need to install the latest version of Visual Studio
gyp ERR! find VS including the "Desktop development with C++" workload.
gyp ERR! find VS For more information consult the documentation at:
gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
gyp ERR! find VS **************************************************************
gyp ERR! find VS
gyp ERR! configure error
gyp ERR! stack Error: Could not find any Visual Studio installation to use
gyp ERR! stack     at VisualStudioFinder.fail (C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:121:47)
gyp ERR! stack     at C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:74:16
gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:351:14)
gyp ERR! stack     at C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:70:14
gyp ERR! stack     at C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:372:16
gyp ERR! stack     at C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\util.js:54:7
gyp ERR! stack     at C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\util.js:33:16
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:315:5)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at maybeClose (internal/child_process.js:1048:16)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Owner\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--module=D:\\Lambda\\projects\\greg\\chaqar-data\\node_modules\\sqlite3\\lib\\binding\\napi-v6-win32-x64\\node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=D:\\Lambda\\projects\\greg\\chaqar-data\\node_modules\\sqlite3\\lib\\binding\\napi-v6-win32-x64" "--napi_version=7" "--node_abi_napi=napi" "--napi_build_version=6" "--node_napi_label=napi-v6"
gyp ERR! cwd D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3
gyp ERR! node -v v14.15.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --module=D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3\lib\binding\napi-v6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3\lib\binding\napi-v6-win32-x64 --napi_version=7 --node_abi_napi=napi --napi_build_version=6 --node_napi_label=napi-v6' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (D:\Lambda\projects\greg\chaqar-data\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1048:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internanode-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1048:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (interdanal/child_process.js:288:5)                                              gy
node-pre-gyp ERR! System Windows_NT 10.0.17134
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "D:\\Lamblida\\projects\\greg\\chaqar-data\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3
node-pre-gyp ERR! node -v v14.15.1                                       a\
node-pre-gyp ERR! node-pre-gyp -v v0.11.0                                ur
node-pre-gyp ERR! not ok                                                 od
Failed to execute 'C:\Program Files\nodejs\node.exe C:\Users\Owner\AppDate=a\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js confesigure --fallback-to-build --module=D:\Lambda\projects\greg\chaqar-data\noapde_modules\sqlite3\lib\binding\napi-v6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=D:\Lambda\projects\greg\chaqar-data\nolede_modules\sqlite3\lib\binding\napi-v6-win32-x64 --napi_version=7 --node_abi_napi=napi --napi_build_version=6 --node_napi_label=napi-v6' (1)      ev
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.1 (node_moduarles\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
                                                                         `
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sqlite3@5.0.1 install: `node-pre-gyp install --fallback-to-build`                                                                        na
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sqlite3@5.0.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additio_0nal logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Owner\AppData\Roaming\npm-cache\_logs\2021-01-11T18_05_27_207Z-debug.log

**Comment by iHildy** [comment]
I am getting the same error. I very well could be wrong but https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v5.0.1/napi-v6-linux-x64.tar.gz does not seem to work.

**Comment by thirionlogan** [comment]
So far, rolling back to 5.0.0 Worked for me

**Comment by ed789d0** [comment]
#1421 is the same issue as this, I believe.

**Comment by iHildy** [comment]
> So far, rolling back to 5.0.0 Worked for me

What command did you do to install the older version?

**Comment by bajtos** [comment]
We are observing errors when installing `sqlite3@5.0.1` on Windows in AppVeyor.  Previous versions worked fine.

Cross-posting from https://github.com/strongloop/loopback-next/pull/7009#issuecomment-758828528:

```
lerna ERR! npm ci exited 1 in '@loopback/typeorm'
lerna ERR! npm ci stdout:
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  nothing.c
  win_delay_load_hook.cc
  nothing.vcxproj -> C:\projects\loopback-next\extensions\typeorm\node_modules\sqlite3\build\Release\\nothing.lib
  unpack_sqlite_dep
  sqlite3.c
  win_delay_load_hook.cc
  sqlite3.vcxproj -> C:\projects\loopback-next\extensions\typeorm\node_modules\sqlite3\build\Release\\sqlite3.lib
  backup.cc
  database.cc
..\src\database.cc(35): error C2248: 'Napi::Env::DefaultFini': cannot access private member declared in class 'Napi::Env' [C:\projects\loopback-next\extensions\typeorm\node_modules\sqlite3\build\node_sqlite3.vcxproj]
  c:\projects\loopback-next\extensions\typeorm\node_modules\node-addon-api\napi-inl.h(475): note: see declaration of 'Napi::Env::DefaultFini'
  C:\projects\loopback-next\extensions\typeorm\node_modules\node-addon-api\napi.h(168): note: see declaration of 'Napi::Env'
  node_sqlite3.cc
  statement.cc
  win_delay_load_hook.cc
Failed to execute 'C:\Program Files (x86)\nodejs\node.exe C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js build --fallback-to-build --module=C:\projects\loopback-next\extensions\typeorm\node_modules\sqlite3\lib\binding\napi-v6-win32-ia32\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\projects\loopback-next\extensions\typeorm\node_modules\sqlite3\lib\binding\napi-v6-win32-ia32 --napi_version=7 --node_abi_napi=napi --napi_build_version=6 --node_napi_label=napi-v6' (1)
```

**Comment by thirionlogan** [comment]
> > So far, rolling back to 5.0.0 Worked for me
> 
> What command did you do to install the older version?

@iHildy I manually changed my package.json from `"sqlite3": "5.0.1",` to `"sqlite3": "5.0.0",` and ran `npm install`

**Comment by iHildy** [comment]
> > > So far, rolling back to 5.0.0 Worked for me
> > 
> > What command did you do to install the older version?
> 
> @iHildy I manually changed my package.json from `"sqlite3": "5.0.1",` to `"sqlite3": "5.0.0",` and ran `npm install`

Yea I did `npm install sqlite3@5.0.0`

**Comment by D19cannon** [comment]
Had te same problem. v5.0.0 worked for me.

**Comment by mcasto** [comment]
Ditto. Rolling it back to 5.0.0 worked for me too.

**Comment by rijkvanzanten** [comment]
@kewde This is a pretty big issue, preventing installing SQLite on numerous machines.. See https://github.com/mapbox/node-sqlite3/issues/1413 as well (seems to be a duplicate)

**Comment by bfate7** [comment]
Same issue , But rolling it back to 5.0.0 worked fine.

**Comment by andrew-barnett** [comment]
same issue #1389

**Comment by shaniefrater** [comment]
rolling back to 5.0.0 worked perfectly

**Comment by zzzkasper-1995** [comment]
For ubuntu 20.04.
it helped me to solve the problem

sudo apt install python
sudo apt install node-gyp
sudo apt upgrade node-gyp
npm install —build-from-source —python=/usr/bin/python2

**Comment by eramdox** [comment]
I found this: https://github.com/mapbox/node-sqlite3/issues/1420

**Comment by jpasi** [comment]
This version works, try!
 npm i sqlite3@5.0.0

**Comment by daliusd** [comment]
Basically the problem is that there are no prebuilt binaries for many platforms and if you don't have built tool installed on your machine you must use older version.

Did someone forgot to run `npx node-pre-gyp publish` or maybe there is process that must do that automatically and has failed to do that?

**Comment by nickrum** [comment]
The buildbots should have published the binaries with 833f5cc1332b9213a17873f42a3d3a18dca5afc1, but that failed due to a recent change. This should hopefully be fixed once #1426 is merged and someone has retriggered the buildbots.

**Comment by JvegaG** [comment]
> > > So far, rolling back to 5.0.0 Worked for me
> > 
> > 
> > What command did you do to install the older version?
> 
> @iHildy I manually changed my package.json from `"sqlite3": "5.0.1",` to `"sqlite3": "5.0.0",` and ran `npm install`

this solution works for me. Thanks 👍

**Comment by jdeltoro88** [comment]
> So far, rolling back to 5.0.0 Worked for me

how did you install 5.0.0

**Comment by 13banda** [comment]
@jdeltoro88 `npm install sqlite3@5.0.0` working for me

**Comment by DJviolin** [comment]
Rolling back to `sqlite3@5.0.0` on Windows also worked for me.

**Comment by yangvaz** [comment]
I had the same issue. 
Rolling back to sqlite3@5.0.0 worked fine for me, thank god

**Comment by kewde** [comment]
I don't have direct access to the AWS S3 bucket so I can not manually edit the files. I have however attempted to rebuild the v5.0.1 which should add the missing N-API v6 binaries for the v5.0.1 release.

**Comment by daliusd** [comment]
Release v5.0.2 has binaries 🥳

**Comment by endiesworld** [comment]
hi, do 
`npm uninstall node-pre-gyp --save
npm install @mapbox/node-pre-gyp --save`
`npm install sqlite3`

the above solved my issue.

**Comment by maitrungduc1410** [comment]
downgrade to 5.0.0 is not working for me, I have to downgrade to 4.1.1

**Comment by yokinzzz** [comment]
downgrade is not working for me

**Comment by daliusd** [comment]
All versions 5.0.0, 5.0.1 and 5.0.2 are working. The only problem with 5.0.1 is that it does not have binaries but they can be build during install phase if you have built tools installed.

If you have problem with 5.x version then the problem most probably is in your side.

**Comment by deepahn** [comment]
This is a network issue and related to proxy setting. Please check proxy setting. I realized my Proxy setting was not working after checking that extensions in VScode was not loading. Now everything is working smoothly!

Strangely the issue was also failing after step > node-pre-gyp install --fallback-to-build. 
After running commands , installing the packages separately, none of them worked. 

Later realized this issue happen because the proxy is not set correctly. I removed the environment variable http_proxy and https_proxy. after that the npm install had no issue running command node-pre-gyp install --fallback-to-build and downloading dependency.

**Comment by Lone-Wolf17** [comment]
> For ubuntu 20.04.
> it helped me to solve the problem
> 
> sudo apt install python
> sudo apt install node-gyp
> sudo apt upgrade node-gyp
> npm install —build-from-source —python=/usr/bin/python2

As at Jul 2021 this solved the problem for me

**Comment by MikeDimmickMnetics** [comment]
If trying to build 5.0.1 from source on Windows, you will need to use Visual Studio 2017 or higher. I got the same error as @bajtos [above](https://github.com/mapbox/node-sqlite3/issues/1424#issuecomment-758831541) when using Visual Studio 2015.

Ensure that you have VS2017 or 2019 C++ build tools available - use Add Or Remove Programs, and click Modify, to see which Visual Studio components you have installed. To force Node to build with the later tools, run:

```
npm config set msvs_version 2017
```

or

```
npm config set msvs_version 2019
```

as appropriate.

**Comment by CleitonGuilherme** [comment]
> Esta é a mensagem de erro completa que estou recebendo ao tentar instalar o sqlite3.
> 
> $ npm e sqlite3
> 
> > sqlite3@5.0.1 install D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3 
> > node-pre-gyp install --fallback-to-build
> 
> node-pre-gyp WARN Usando solicitação para node-pre-gyp https download node-pre-gyp WARN Tentativa de download (403): [https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v5.0.1 /napi-v6-win32-x64.tar.gz](https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v5.0.1/napi-v6-win32-x64.tar.gz) node-pre-gyp WARN Binários pré-criados não encontrados para sqlite3@5.0.1 e node@14.15.1 (node-v83 ABI, desconhecido) (recorrendo à compilação de origem com node-gyp) gyp ERR! encontrar VS gyp ERR! encontre VS msvs_version não definido na linha de comando ou npm config gyp ERR! encontre o VS VCINSTALLDIR não definido, não sendo executado no VS Command Prompt gyp ERR! find VS não pôde usar o PowerShell para encontrar o Visual Studio 2017 ou gyp ERR mais recente! encontre o VS procurando pelo Visual Studio 2015 gyp ERR! encontrar VS - não encontrado gyp ERR! encontre o VS não procurando pelo VS2013, pois ele só é suportado até Node.js 8 gyp ERR! encontrar VS gyp ERR! encontrar VS ************************************************** ************** gyp ERR! find VS Você precisa instalar a versão mais recente do Visual Studio gyp ERR! encontre o VS incluindo a carga de trabalho "Desktop development with C++". gyp ERR! find VS Para mais informações consulte a documentação em: gyp ERR! encontre VS https://github.com/nodejs/node-gyp#on-windows gyp ERR! encontrar VS ************************************************** ************** gyp ERR! encontrar VS gyp ERR! configurar erro gyp ERR! Erro de pilha: não foi possível encontrar nenhuma instalação do Visual Studio para usar gyp ERR! pilha em VisualStudioFinder.fail (C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:121:47) gyp ERR! pilha em C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:74:16 gyp ERR! pilha em VisualStudioFinder.findVisualStudio2013 (C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:351:14) gyp ERR! pilha em C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:70:14 gyp ERR! pilha em C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:372:16 gyp ERR! pilha em C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\util.js:54:7 gyp ERR! pilha em C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\util.js:33:16 gyp ERR! pilha em ChildProcess.exithandler (child_process.js:315:5) gyp ERR! pilha em ChildProcess.emit (events.js:315:20) gyp ERR! pilha em mayClose (internal/child_process.js:1048:16) gyp ERR! Sistema Windows_NT 10.0.17134 gyp ERR! command "C:\Program Files\nodejs\node.exe" "C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" "configure" " --fallback-to-build" "--module=D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3\lib\binding\napi-v6-win32-x64\node_sqlite3.node" "--module_name =node_sqlite3" "--module_path=D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3\lib\binding\napi-v6-win32-x64" "--napi_version=7" "--node_abi_napi=napi " "--napi_build_version=6" "--node_napi_label=napi-v6" gyp ERR! cwd D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3 gyp ERR! nó -v v14.15.1 gyp ERR! node-gyp -v v5.1.0 gyp ERR! não ok nó-pre-gyp ERR! nó-pré-gyp ERR! Erro de pilha: Falha ao executar 'C:\Program Files\nodejs\node.exe C:\Users\Owner\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure - -fallback-to-build --module=D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3\lib\binding\napi-v6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path =D:\Lambda\projects\greg\chaqar-data\node_modules\sqlite3\lib\binding\napi-v6-win32-x64 --napi_version=7 --node_abi_napi=napi --napi_build_version=6 --node_napi_label=napi- v6' (1) nó-pre-gyp ERR! pilha em ChildProcess. (D:\Lambda\projects\greg\chaqar-data\node_modules\node-pre-gyp\lib\util\compile.js:83:29) node-pre-gyp ERR! pilha em ChildProcess.emit (events.js:315:20) nó-pré-gyp ERR! empilhar em mayClose (internal/child_process.js:1048:16) node-pre-gyp ERR! pilha em Process.ChildProcess._handle.onexit (internanode-pre-gyp ERR! pilha em mayClose (internal/child_process.js:1048:16) node-pre-gyp ERR! pilha em Process.ChildProcess._handle.onexit (interdanal/ child_process.js:288:5) gy node-pre-gyp ERR! Sistema Windows_NT 10.0.17134 node-pre-gyp ERR! comando "C:\Program Files\nodejs\node.exe" "D:\Lamblida\projects\ greg\chaqar-data\node_modules\node-pre-gyp\bin\node-pre-gyp" "instalar" "--fallback-to-build" node-pre-gyp ERR! cwd D:\Lambda\projects\greg \chaqar-data\node_modules\sqlite3 node-pre-gyp ERR! node -v v14.15. nó-pré-gyp ERR! node-pre-gyp -v v0.11.0 ur node-pre-gyp ERR! not ok od Falha ao executar 'C:\Program Files\nodejs\node.exe C:\Users\Owner\AppDate=a\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js confisigure --fallback-to-build --module=D:\Lambda\projects\greg\chaqar-data\noapde_modules\sqlite3\lib\binding\napi-v6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 - -module_path=D:\Lambda\projects\greg\chaqar-data\nolede_modules\sqlite3\lib\binding\napi-v6-win32-x64 --napi_version=7 --node_abi_napi=napi --napi_build_version=6 --node_napi_label= napi-v6' (1) ev npm WARN opcional PULAR DEPENDÊNCIA OPCIONAL: fsevents@2.3.1 (node_moduarles\fsevents): npm WARN notsup PULAR DEPENDÊNCIA OPCIONAL: plataforma não suportada para fsevents@2.3.1: queria {"os":"darwin","arch":"any"} (atual: {"os":"win32","arch": "x64"}) `npm ERR! code ELIFECYCLE npm ERR! errno 1 npm ERR! sqlite3@5.0.1 install:`node-pre-gyp install --fallback-to-build` na npm ERR! Status de saída 1 npm ERR! npm ERR! Falha no script de instalação sqlite3@5.0.1. npm ERR! Isso provavelmente não é um problema com o npm. Provavelmente há saída de registro adicional acima.
> 
> npm ERR! Um log completo desta execução pode ser encontrado em: npm ERR! C:\Users\Proprietário\AppData\Roaming\npm-cache_logs\2021-01-11T18_05_27_207Z-debug.log

mano no meu para poder resolver o erro foi só instalar o pyton que eu ainda não tinha na maquina, tente isso, acredito que vai da certo., caso contrario, remova o node_modules e package.son e depois npm i acredito que vai resolver, porém tente a primeira opção.

**Comment by raja-muhammad-asher** [comment]
> 

Solved my issue for Ubuntu 22.04 and version 5.0.2

**Comment by craverath** [comment]
I love u
